src/main/java/com/uipath/uipathpackage/util/TaskScopedEnvVarsManager.java

```diff
@@ -0,0 +1,19 @@
+package com.uipath.uipathpackage.util;
+
+import hudson.EnvVars;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+
+public class TaskScopedEnvVarsManager {
+    public static EnvVars selectOnlyRequiredEnvironmentVariables(@Nonnull Run<?, ?> run, @Nonnull EnvVars env, @Nonnull TaskListener listener) throws IOException, InterruptedException {
```

Comment:
Maybe change the name of the method to addRequiredEnvironmentVariables. The older name seems to restrict the environment variables that we will use.